### PR TITLE
[REVIEW ONLY] PP-1753 Upgraded project dependencies

### PR DESCRIPTION
**Pull request by ckalista**
## WHAT
_A brief description of the pull request:_

- Upgraded the following libraries:
  - `Dropwizard` from version `0.8.1` to `1.0.6`.
  - `commons-validator` from version `1.5.1` to `1.6`.
  - `io.dropwizard.metrics` from version `3.1.2` to `3.2.0`.
  - `com.fasterxml.jackson.datatype` from version `2.8.6` to `2.8.7`.
  - `com.fasterxml.jackson.core` from version `2.8.6` to `2.8.7`.
  - `black.door` from version `v1r3t1` to `v1r4t0`
  - `httpclient` from version `4.5.2` to `4.5.3`
  - `jersey-common` from version `2.23.2` to `2.25.1`
  - `jersey-media-jaxb` from version `2.23.2` to `2.25.1`

- The implementation of `Dropwizard Authentication` has undergone a major overhaul since version 0.9.x of `Dropwizard` introducing a new interface for `Authenticators` and the use of a `Principal`. The code was refactored to conform to the new Auth implementation.

- Added test checking that colon characters `:` that can be found in the query string portion of all navigation links are being encoded as expected since the upgrade of Jersey library. 

- The `Dropwizard` upgrade caused a dependency conflicts with `ch.qos.logback` which was resolved by pulling the latest version explicitly.

NOTE: The SourceClear report still shows a medium risk vulnerabilites in `jersey-common` and `jersey-media-jaxb` libraries even in the latest version.

https://app.sourceclear.com/teams/100t11M/scans/1050886

### CODE REVIEW

**Comment by ckalista** [comment]
rebuild

File: pom.xml [hunk 1]
```diff
@@ -42,12 +42,32 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.core</groupId>
+            <artifactId>jersey-common</artifactId>
+            <version>2.25.1</version>
```
Comment (alexbishop1) [inline pom.xml]: Can use `<version>${jersey2.version}</version>` here?

File: pom.xml [hunk 2]
```diff
@@ -42,12 +42,32 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.media</groupId>
+            <artifactId>jersey-media-jaxb</artifactId>
+            <version>2.25.1</version>
```
Comment (alexbishop1) [inline pom.xml]: Another opportunity for `<version>${jersey2.version}</version>`

File: pom.xml [hunk 3]
```diff
@@ -42,12 +42,32 @@
...
+            <version>3.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
```
Comment (alexbishop1) [inline pom.xml]: Why was this added? I assume it’s explicitly replacing a previously-transient dependency but I don’t see anything about it in the pull request.

Comment (ckalista) [inline pom.xml]: `httpclient` is an indirect project dependency. SourceClear reported that version `4.5.2` needed to be upgraded specifically to `4.5.3`.

File: pom.xml (hunk 1 above)
Comment (ckalista) [inline pom.xml]: Indeed! Suggestion applied.

File: pom.xml (hunk 2 above)
Comment (ckalista) [inline pom.xml]: Suggestion applied.

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild
